pkg/endpoint/local_ip.go

```diff
@@ -36,14 +41,18 @@ func GetLocalIPForDestination(dst string) string {
 	return localAddr.IP.String()
 }
 
-func GetLocalIP(family k8snet.IPFamily) string {
+func GetLocalIPForDest(dst string, family k8snet.IPFamily) string {
```

Comment:
I think the names `GetLocalIPForDest` and `GetLocalIPForDestination` are a bit ambiguous wrt to their purpose. As I suggested earlier, I think we should name this `GetLocalIPForDestination`. The prior `GetLocalIPForDestination` on L32 is only for IPv4 (currently anyway) so I think we should name it accordingly. Also I don't think we need to expose it publicly, hence `getLocalIPv4ForDestination`.

Reply:
Done